src/app/utils/field-validation/validators/dateValidator.ts

```diff
@@ -111,27 +111,17 @@ const makeDateValidator: DateValidatorConstructor = (dateField) => {
 }
 
 /**
- * Returns the number representation of a day of the week
+ * constant
  */
-const convertInvalidDayToNumber = (invalidDay: InvalidDaysOptions): number => {
-  switch (invalidDay) {
-    case InvalidDaysOptions.Sunday:
-      return 0
-    case InvalidDaysOptions.Monday:
-      return 1
-    case InvalidDaysOptions.Tuesday:
-      return 2
-    case InvalidDaysOptions.Wednesday:
-      return 3
-    case InvalidDaysOptions.Thursday:
-      return 4
-    case InvalidDaysOptions.Friday:
-      return 5
-    case InvalidDaysOptions.Saturday:
-      return 6
-    default:
-      return -1
-  }
+const INVALID_DAY_MAP: Record<InvalidDaysOptions, number> = {
+  [InvalidDaysOptions.Sunday]: 0,
+  [InvalidDaysOptions.Monday]: 1,
+  [InvalidDaysOptions.Tuesday]: 2,
+  [InvalidDaysOptions.Wednesday]: 3,
+  [InvalidDaysOptions.Thursday]: 4,
+  [InvalidDaysOptions.Friday]: 5,
+  [InvalidDaysOptions.Saturday]: 6,
+  [InvalidDaysOptions.SingaporePublicHolidays]: 7,
```

Comment:
was this not already removed? this should not be `7` regardless imo. this is to be used for getting the day of the week. Maybe a better name for the constant could be `DAY_TO_NUMBER_MAP`.

Reply:
Awaiting to merge PR to remove Singapore public holidays before I can rebase this branch

Reply:
Updated in latest commit